### PR TITLE
XDR speed decoding support 

### PR DESCRIPTION
**Pull request by gregoryl-mlnx**
This PR add support XDR speed decoding to the:
 - libibumad 
 - infinabnd-diags
 - ibnetdiscover
 - iblinkinfo
 - ibqueryerrors
 - ibportstate

### CODE REVIEW

**Comment by rleon** [comment]
XDR is not in final IBTA spec version yet.

**Comment by gregoryl-mlnx** [comment]
> XDR is not in final IBTA spec version yet.

Yes, but I feel like format of XDR speed encoding/decoding will not be changed.
In any case it's a initial PR, we need it for internal testing and integration with other components.
We should not must merge it today, if you see risks.

File: libibnetdisc/libibnetdisc.map [hunk 1]
```diff
@@ -18,5 +18,13 @@ IBNETDISC_1.0 {
...
 		ibnd_find_port_lid;
 		ibnd_iter_ports;
+		ibnd_get_agg_linkspeedext_field;
+		ibnd_get_agg_linkspeedext;
+		ibnd_get_agg_linkspeedexten;
+		ibnd_get_agg_linkspeedextsup;
+		ibnd_dump_agg_linkspeedext_bits;
+		ibnd_dump_agg_linkspeedext;
+		ibnd_dump_agg_linkspeedexten;
+		ibnd_dump_agg_linkspeedextsup;
```
[the remark this answers is not in the document]
Comment (gregoryl-mlnx) [reply libibnetdisc/libibnetdisc.map:33]: fixed

File: libibnetdisc/libibnetdisc.map [hunk 2]
```diff
@@ -20,3 +20,16 @@ IBNETDISC_1.0 {
...
+	global:
+		ibnd_get_agg_linkspeedext_field;
+		ibnd_get_agg_linkspeedext;
+		ibnd_get_agg_linkspeedexten;
+		ibnd_get_agg_linkspeedextsup;
+		ibnd_dump_agg_linkspeedext_bits;
+		ibnd_dump_agg_linkspeedext;
+		ibnd_dump_agg_linkspeedexten;
+		ibnd_dump_agg_linkspeedextsup;
+	local: *;
```
Comment (rleon) [inline libibnetdisc/libibnetdisc.map:34]: This line is not needed, but ok

**Pull request merged**